src/screens/splash.rs

```diff
@@ -35,48 +39,82 @@ pub(super) fn plugin(app: &mut App) {
         )
             .run_if(in_state(Screen::Splash)),
     );
+
+    // Exit the splash screen early if the player hits escape.
+    app.add_systems(
+        Update,
+        exit_splash_screen
+            .run_if(input_just_pressed(KeyCode::Escape).and_then(in_state(Screen::Splash))),
+    );
 }
 
 const SPLASH_BACKGROUND_COLOR: Color = Color::srgb(0.157, 0.157, 0.157);
 const SPLASH_DURATION_SECS: f32 = 1.8;
 const SPLASH_FADE_DURATION_SECS: f32 = 0.6;
 
-fn spawn_splash(mut commands: Commands, asset_server: Res<AssetServer>) {
-    commands
-        .ui_root()
-        .insert((
-            Name::new("Splash screen"),
+#[derive(Component, Reflect)]
+#[reflect(Component)]
+struct SplashScreenContainer;
+
+#[derive(Resource, Reflect)]
+#[reflect(Resource)]
+struct SplashScreenImageList(VecDeque<Handle<Image>>);
+
+impl FromWorld for SplashScreenImageList {
+    fn from_world(world: &mut World) -> Self {
+        let asset_server = world.resource::<AssetServer>();
+
+        // To show your own splash images, replace or add images here.
+        Self(VecDeque::from_iter(
+            [asset_server.load("images/splash.png")],
+        ))
+    }
+}
+
+fn exit_splash_screen(mut next_screen: ResMut<NextState<Screen>>) {
+    next_screen.set(Screen::Loading);
+}
+
+fn splash_image_bundle(image: Handle<Image>) -> impl Bundle {
+    let image = image.into();
+
+    (
+        Name::new("Splash image"),
+        ImageBundle {
+            style: Style {
+                margin: UiRect::all(Val::Auto),
+                width: Val::Percent(70.0),
+                ..default()
+            },
+            image,
+            background_color: BackgroundColor(Color::srgba(0., 0., 0., 0.)),
+            ..default()
+        },
+        UiImageFadeInOut {
+            total_duration: SPLASH_DURATION_SECS,
+            fade_duration: SPLASH_FADE_DURATION_SECS,
+            t: 0.0,
+        },
+    )
+}
+
+fn spawn_splash(
+    mut commands: Commands,
+    splash_images: Res<SplashScreenImageList>,
+    mut next_screen: ResMut<NextState<Screen>>,
+) {
+    if splash_images.0.is_empty() {
```

Comment:
Don't think this check is necessary. Doesn't the system with the timer do this for us anyways?

Reply:
This has been removed, but this prevents a pause for the timer's duration with a blank screen if no images are provided.